src/contrib/cppheaderparser/CppHeaderParser/CppHeaderParser.py

```diff
@@ -1879,7 +1879,7 @@ def evaluate_property_stack(self):
                 typedef = self._parse_typedef( self.stack )
                 name = typedef['name']
                 klass = self.classes[ self.curClass ]
-                klass[ 'typedefs' ][ self.curAccessSpecifier ].append( name )
+                klass[ 'typedefs' ][ self.curAccessSpecifier ].append( typedef )
```

Comment:
@lalvarezt Did you find any issue with CppHeaderParser ? I think you should report the issue as well as your fix, I don't want our version to diverge with upstream. I didn't check https://bitbucket.org/senex/cppheaderparser for a while, maybe this fix is already available ?